src/ButteredBread.sol

```diff
@@ -115,7 +115,8 @@ contract ButteredBread is ERC20VotesUpgradeable, OwnableUpgradeable, IButteredBr
         _accountToLPData[_account][_lp].scalingFactor = currentScalingFactor;
 
         _mint(_account, _amount * currentScalingFactor / FIXED_POINT_PERCENT);
-        if (this.delegates(_account) == address(0)) _delegate(_account, _account);
+        if (this.delegates(_account) != address(0)) _delegate(_account, this.delegates(_account));
```

Comment:
essentially here we need to check if in the $BREAD contract (forgot to mention that in the issue, we need to add that contract here) the user is delegated to someone. if they are delegated to someone, we need to delegate their $BB balance to that user as well. So if `if (BREAD.delegates(_account) != address(0)) _delegate(_account, BREAD.delegates(_account))`